types/resolve/index.d.ts

```diff
@@ -113,17 +121,21 @@ declare namespace resolve {
     isDirectory?: (directory: string, cb: existsCallback) => void;
     /** function to asynchronously resolve a potential symlink to its real path */
     realpath?: (file: string, cb: realpathCallback) => void;
+    /** function to asynchronously read and parse a package.json file */
+    readPackage?: (readFile: (file: string, cb: readFileCallback) => void, pkgfile: string, cb: readPackageCallback) => void;
   }
 
   export interface SyncOpts extends Opts {
     /** how to read files synchronously (defaults to fs.readFileSync) */
-    readFileSync?: (file: string, encoding: BufferEncoding) => string | Buffer;
```

Comment:
`encoding` is not passed: https://github.com/browserify/resolve/blob/2e2d2630122a8e6a3205f9a8baa5be50ab7b3108/lib/sync.js#L49